### PR TITLE
Prepare to be gcc7 ready

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: lib/libzlib/sub.mk [hunk 1]
```diff
@@ -1,11 +1,11 @@
 global-incdirs-y += include
 
+cflags-y += -Wno-extra
 cflags-y += -Wno-old-style-definition
 cflags-y += -Wno-switch-default
 cflags-y += -Wno-strict-prototypes
 cflags-y += -Wno-missing-prototypes
 cflags-y += -Wno-missing-declarations
-cflags-y += -Wno-shift-negative-value
```
Comment (igoropaniuk) [inline lib/libzlib/sub.mk]: Are warnings about shifting negative values  disabled by default in GCC7? (or I guess they are included in extra warning, right?)

Comment (jenswi-linaro) [inline lib/libzlib/sub.mk]: Yes, they are part of `-Wextra`

**Comment by igoropaniuk** [comment]
`Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>`

**Comment by jenswi-linaro** [comment]
Tags applied

**Pull request merged**